### PR TITLE
Remove the counting of connections

### PR DESCRIPTION
**Pull request by fclairamb**
# Goal
I'd like to remove everything that is not strictly necessary to implement a FTP server. The check around the maximum number of connections, if necessary, should be done on the server implementation itself.

# Changes
- Removed all codes that counts the number of connections. This allowed to remove the last reference to the sync package.
- Implemented this part on the sample driver code itself

### CODE REVIEW

**Review by mgenov — APPROVED** [review]
LGTM !

**Review by asv — APPROVED** [review]
Looks good.

File: server/server.go [hunk 1]
```diff
@@ -187,29 +181,19 @@ func (server *FtpServer) Stop() {
 }
 
 // When a client connects, the server could refuse the connection
-func (server *FtpServer) receiveConnection(conn net.Conn) error {
-	nb := int(atomic.AddInt32(&server.clientsNb, 1))
-	id := atomic.AddUint32(&server.clientCounter, 1)
+func (server *FtpServer) clientArrival(conn net.Conn) error {
+	server.clientCounter++
+	id := server.clientCounter
```
Comment (awskii) [inline server/server.go:186]: Situation when two clients will get same id is not excluded.

Comment (fclairamb) [reply server/server.go:186]: clientArrival was previously called on the client goroutine but now it's on the listening goroutine. This count can't be called by two goroutines at the same time.

**Pull request merged**